src/System.Memory/src/System/Buffers/ArrayMemoryPool.cs

```diff
@@ -22,7 +22,9 @@ public sealed override OwnedMemory<T> Rent(int minimumBufferSize = -1)
             else if (((uint)minimumBufferSize) > s_maxBufferSize)
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.minimumBufferSize);
 
-            return new ArrayMemoryPoolBuffer(minimumBufferSize);
+            var buffer = new ArrayMemoryPoolBuffer(minimumBufferSize);
+            buffer.Retain();
```

Comment:
This doesn't need to incur an interlocked operation, as it's not published yet.

Reply:
Do we want to do this? It changes the contract quite a bit. Can you fix pipelines in the same PR?

Reply:
I'll fix pipelines. It's just strange to get an OwnedMemory with 0 references.

Reply:
Everyone has to agree on the semantics, today our pool in kestrel does not do this so it'll result in memory leaks...

Reply:
I removed Retain and added dispose call to Pipelines.